CorsixTH/Lua/entities/humanoids/staff.lua

```diff
@@ -578,16 +583,20 @@ end
 -- any request raise dialogs.
 --!param amount (integer) The amount, in game dollars per month, to increase
 -- the salary by.
+--!return (bool) Whether wage increase was successful
 function Staff:increaseWage(amount)
-  self.profile.wage = self.profile.wage + amount
-  self.world.ui:playSound("cashreg.wav")
-  if self.profile.wage > 2000 then -- What cap here?
-    self.profile.wage = 2000
-  else -- If the cap has been reached this member of staff won't get unhappy
-       -- ever again...
+  -- Are we already paid the maximum?
+  if self.profile.wage >= self.max_salary then return false end
+  if self.profile.wage + amount > self.max_salary then
+    -- Maximum salary hit. The staff member will never be unhappy again
+    self.profile.wage = self.max_salary
+  else
+    -- Apply new salary
+    self.profile.wage = self.profile.wage + amount
     self:changeAttribute("happiness", 0.99)
     self:setMood("pay_rise", "deactivate")
```

Comment:
Wonder if we should always be running this line? I think this might've been written wrong.

Reply:
Outcome: yes